wp-content/plugins/core/src/Templates/Models/Post_List_Object.php

```diff
@@ -11,72 +11,135 @@ class Post_List_Object {
 	public const LINK      = 'link';
 	public const POST_TYPE = 'post_type';
 
-	private int $post_id;
-	private string $title;
-	private string $content;
-	private string $excerpt;
-	private int $image_id;
-	private array $link;
-	private string $post_type;
-
-	public function __construct( array $args = [] ) {
-		$this->post_id   = (int) $args[ self::POST_ID ] ?? 0;
-		$this->title     = (string) $args[ self::TITLE ] ?? '';
-		$this->content   = (string) $args[ self::CONTENT ] ?? '';
-		$this->excerpt   = (string) $args[ self::EXCERPT ] ?? '';
-		$this->image_id  = (int) $args[ self::IMAGE ] ?? 0;
-		$this->link      = (array) $args[ self::LINK ] ?? [];
-		$this->post_type = (string) $args[ self::POST_TYPE ] ?? '';
-	}
+	/**
+	 * @var int
+	 */
+	private $post_id;
+
+	/**
+	 * @var string
+	 */
+	private $title;
+
+	/**
+	 * @var string
+	 */
+	private $content;
+
+	/**
+	 * @var string
+	 */
+	private $excerpt;
+
+	/**
+	 * @var int
+	 */
+	private $image_id;
+
+	/**
+	 * @var array
+	 */
+	private $link;
+
+	/**
+	 * @var string
+	 */
+	private $post_type;
```

Comment:
What was the reasoning behind removing all the type declarations?

Reply:
Great question. Because of the uninitialized state, when I check if the object "is valid" the get_title() methods throw errors. I felt this was cleaner/easier? Wondering your thoughts? Maybe the getters do something like this
'''
public function get_title() {
    return isset($this-title) ? $this-title : '';
}
'''

Reply:
Why not initialize the variables when they're declared, e.g. `private string $title = '';` ?

Reply:
Yep, good idea. It's not a pattern we've followed for the components is the only reason I didn't do this but this is a separate thing.

Reply:
Don't forget to add the return types back to the methods :)